concent_api/core/templatetags/admin_tags.py

```diff
@@ -0,0 +1,52 @@
+import os
+from django import template
+from constance import config
+from ..models import Subtask
+
+register = template.Library()
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+
+@register.assignment_tag
+def base_dir_path():
+    return BASE_DIR
+
+
+@register.assignment_tag
+def is_shutdown_mode_enabled():
+    return config.SOFT_SHUTDOWN_MODE
+
+
+def result_uploaded_subtasks():
+    return [subtask for subtask in all_subtasks() if subtask.state == 'RESULT_UPLOADED']
+
+
+@register.assignment_tag
+def active_subtasks_amount():
+    return len(active_subtasks())
+
+
+@register.assignment_tag
+def result_uploaded_subtasks_amount():
+    return len(result_uploaded_subtasks())
+
+
+@register.assignment_tag
+def active_subtasks():
+    active_state_names = [state.name for state in Subtask.ACTIVE_STATES]
+    return [subtask for subtask in all_subtasks() if subtask.state in active_state_names]
+
+
+@register.assignment_tag
+def all_subtasks():
+    return Subtask.objects.all()
+
+
+@register.assignment_tag
+def shutdown_mode_state():
+    if active_subtasks_amount() > 0:
```

Comment:
1. Don't call `active_subtasks_amount` more than once, assign result to the variable and reuse it. 
2. Don't use hardcoded strings, I think in this case it would be best to make a separate tag for each condition.

Reply:
Done.